src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -29,13 +33,17 @@ class ArrayAdapter implements AdapterInterface, CacheInterface, LoggerAwareInter
     private $values = [];
     private $expiries = [];
     private $createCacheItem;
+    private $maxLifetime;
+    private $maxItems;
 
     /**
      * @param bool $storeSerialized Disabling serialization can lead to cache corruptions when storing mutable values but increases performance otherwise
      */
-    public function __construct(int $defaultLifetime = 0, bool $storeSerialized = true)
+    public function __construct(int $defaultLifetime = 0, bool $storeSerialized = true, int $maxLifetime = 0, int $maxItems = 0)
     {
         $this->storeSerialized = $storeSerialized;
+        $this->maxLifetime = 0 < $maxLifetime ? $maxLifetime : 0;
```

Comment:
Wouldn't it be better to throw an exception? It looks like a silent error recovery to me.

Reply:
works for me, updated